### DISCUSSION TITLE
Forcing Endpoint's sink completion from receiveValue

### DISCUSSION THREAD

**Discussion by biox86** [sbertix/Swiftagram#210] [Q&A] [answered]
With the 4.x version, with a few endpoints, I used to check inside the receiveValue if the returned data was new or if it was already cached from a previous execution of the same endpoint.
In that case I could force the end of the endpoint task calling task?.cancel() and then work on the partial data inside the task completion callback.

Is there a way to force the task completion with ver 5.x?

Remove all the items inside the bin will result in a total cancellation of the task, without prompting the completion callback.

**Comment by sbertix** [comment, marked as answer]
Hey @biox86,

You can use **Combine** `Publisher` `handleEvents` to detect cancellation, etc.
